go/oasis-node/node_test.go

```diff
@@ -50,19 +50,10 @@ var (
 		value interface{}
 	}{
 		{"log.level.default", "DEBUG"},
-		{"epochtime.backend", "tendermint_mock"},
-		{"consensus.backend", "tendermint"},
 		{"consensus.validator", true},
-		{"registry.debug.allow_unroutable_addresses", true},
-		{"registry.debug.allow_runtime_registration", true},
-		{"registry.debug.bypass_stake", true},
 		{"roothash.tendermint.index_blocks", true},
-		{"scheduler.debug.bypass_stake", true},
-		{"scheduler.debug.static_validators", true},
 		{"storage.backend", "leveldb"},
 		{"staking.debug.genesis_state", stakingTests.DebugGenesisState},
```

Comment:
This flag should be folded into `NewSingleNodeGenesisProvider` and gotten rid of somehow.

Reply:
Do we even need a separate staking genesis state? This can probably be removed now as long as there is a (test?) genesis document provider which supplies the right genesis values.

Reply:
Not anymore, though it shouldn't be in `NewSingleNodeGenesisProvider` if we want to continue to support running single node networks.

Reply:
AFAIK the only single node network now is the one in unit tests (now that there is the network runner). So maybe just rename it to `NewTestNodeGenesisProvider` or something and just drop support for running nodes in a single-node-without-genesis-file configuration?

Reply:
I would be ok with that.

Reply:
Weren't we going to remove that flag and just fold the test genesis state into the provider somehow?

Reply:
I'd rather leave this refactor for a separate PR, as I'll be away for the next few days and will have to go through merge-conflict hell when I return if this PR isn't merged soon :)
I did rename the genesis provider and made it available only in unit tests, so it should at least be hard to trigger by accident.

Reply:
Nevermind, I mixed things up, moving the debug staking stuff into the test genesis doc is actually not that much extra work, I'll do it now :)

Reply:
OK, I implemented this in a separate commit, please re-review.